lib/kafka/cluster.rb

```diff
@@ -40,7 +40,7 @@ def initialize(seed_brokers:, broker_pool:, logger:)
     # @param topics [Array<String>]
     # @return [nil]
     def add_target_topics(topics)
-      new_topics = Set.new(topics) - @target_topics
+      new_topics = Set.new(topics).delete_if {|o| @target_topics.include? o }
 
       unless new_topics.empty?
         @logger.info "New topics added to target list: #{new_topics.to_a.join(', ')}"
```

Comment:
I think a better approach, if benchmarking validates it, is to check if `topics` id a subset of `@target_topics`.

```ruby
topics = Set.new(topics)

unless topics.subset?(@target_topics)
  new_topics = Set.new(topics) - @target_topics

  ...
end
```

I would guess that that's fast enough.